From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Charles Kerr <charles@charleskerr.com>
Date: Tue, 4 Aug 2026 18:21:28 -0500
Subject: test: account for libc++ rm error on Electron Linux

Patch a test that attempts a rimraf-style recursive removal on a subtree
where a subdirectory isn't writable. Uses std::filesystem::remove_all(),
which reports `permission_denied` on libstdc++ but `directory_not_empty`
on libc++. Node exposes these as `EACCES` and `ENOTEMPTY.

The test assumes libstdc++ on Linux; but since we're using Chromium's
libc++, we get a different error code. This patch adjusts the test's
expectation.

CI doesn't hit this since we run tests as root and the block only runs
iff `process.getuid() !== 0`. But it does fail if run on Linux by a
user with ordinary permissions.

Refs: https://github.com/nodejs/node/pull/57103

diff --git a/test/parallel/test-fs-rm.js b/test/parallel/test-fs-rm.js
index e92bf8c079715fc97f491f83be09a606769eba18..b15118fc9aa0d63688cdac07e8004f452d86d08e 100644
--- a/test/parallel/test-fs-rm.js
+++ b/test/parallel/test-fs-rm.js
@@ -491,7 +491,8 @@ if (isGitPresent) {
   if (!common.isIBMi && (common.isWindows || process.getuid() !== 0)) {
     function makeDirectoryReadOnly(dir, allowExecute) {
       let accessErrorCode = 'EACCES';
-      if (common.isMacOS && allowExecute) {
+      if (allowExecute &&
+          (common.isMacOS || (common.isLinux && process.versions.electron))) {
         accessErrorCode = 'ENOTEMPTY';
       }
       if (common.isWindows) {
